crates/rbuilder/src/backtest/fetch/csv.rs

```diff
@@ -0,0 +1,113 @@
+use crate::primitives::Order;
+use crate::{
+    backtest::{
+        fetch::datasource::{BlockRef, DataSource},
+        OrdersWithTimestamp,
+    },
+    primitives::{Bundle, TransactionSignedEcRecoveredWithBlobs},
+};
+use alloy_rlp::Decodable;
+use async_trait::async_trait;
+use csv::Reader;
+use eyre::Context;
+use reth::primitives::TransactionSignedEcRecovered;
+use std::{collections::HashMap, fs::File, path::PathBuf};
+use tracing::trace;
+use uuid::Uuid;
+
+#[derive(Debug, Clone)]
+pub struct CSVDatasource {
+    batches: HashMap<u64, Vec<TransactionSignedEcRecovered>>,
+}
+
+impl CSVDatasource {
+    pub fn new(filename: impl Into<PathBuf>) -> eyre::Result<Self> {
+        let batches = Self::load_transactions_from_csv(filename.into())?;
+        Ok(Self { batches })
+    }
+
+    fn load_transactions_from_csv(
+        filename: PathBuf,
+    ) -> eyre::Result<HashMap<u64, Vec<TransactionSignedEcRecovered>>> {
+        let file = File::open(&filename)
+            .wrap_err_with(|| format!("Failed to open file: {}", filename.display()))?;
+        let mut reader = Reader::from_reader(file);
+        let mut batches: HashMap<u64, Vec<TransactionSignedEcRecovered>> = HashMap::new();
+
+        for result in reader.records() {
+            let record = result?;
+            if record.len() != 2 {
```

Comment:
0,1,and 2 could be constants also documenting the format
eg: 
/// col 1 contains bla bla
const COL_RAW_RLP_RX = 1
Is col 0 an arbitrary number? Is this some block number?